### PR TITLE
Expand fields to simplify string functions

### PR DESCRIPTION
**Pull request by snewcomer**
Ref https://github.com/jeregrine/jsonapi/pull/158

@jherdman

### CODE REVIEW

File: lib/jsonapi/utils/string.ex [hunk 1]
```diff
@@ -122,19 +79,25 @@ defmodule JSONAPI.Utils.String do
...
       iex> camelize("_top__posts_")
       "_top__posts_"
 
   """
+  @spec camelize(atom) :: atom
   def camelize(value) when is_atom(value) do
     value
     |> to_string()
     |> camelize()
+    |> String.to_atom()
```
Comment (snewcomer) [inline lib/jsonapi/utils/string.ex]: For outgoing data, what reason would we convert it back to an atom?  `underscore` did this as well

File: lib/jsonapi/utils/string.ex [hunk 2]
```diff
@@ -154,11 +154,12 @@ defmodule JSONAPI.Utils.String do
...
 
+      iex> expand_fields(%{"inserted-at" => ~N[2019-01-17 03:27:24.776957]}, &underscore/1)
+      %{"inserted_at" => ~N[2019-01-17 03:27:24.776957]}
+
   """
   @spec expand_fields(map, function) :: map
-  def expand_fields(%{__struct__: _} = value, _fun) when is_map(value) do
-    value
-  end
+  def expand_fields(%{__struct__: _} = value, _fun), do: value
```
Comment (jherdman) [inline lib/jsonapi/utils/string.ex:160]: I confess that I'm not really sure what's happening with this `__struct__:` thing. What's the gist?

File: lib/jsonapi/utils/string.ex [hunk 3]
```diff
@@ -17,58 +17,29 @@ defmodule JSONAPI.Utils.String do
...
-      %{"f_b" => %{"a_d" => "z"}, "c_d" => "e"}
-
-      iex> underscore(:"foo-bar")
-      :foo_bar
-
-      iex> underscore(%{"f-b" => "a-d"})
-      %{"f_b" => "a-d"}
-
   """
+  @spec underscore(String.t()) :: String.t()
```
Comment (jherdman) [inline lib/jsonapi/utils/string.ex:30]: ❤️

File: lib/jsonapi/utils/string.ex (hunk 2 above)
Comment (snewcomer) [reply lib/jsonapi/utils/string.ex:160]: Pattern matching a naive datetime and other data types - `~N[2019-01-17 03:27:24.776957]`

File: lib/jsonapi/utils/string.ex (hunk 1 above)
Comment (jherdman) [reply lib/jsonapi/utils/string.ex]: IIRC this came from some code review feedback here: https://github.com/jeregrine/jsonapi/pull/149#discussion_r243855630

Comment (doomspork) [reply lib/jsonapi/utils/string.ex]: I think we can ditch the `to_atom/1` calls, I'm not sure that's a good habit to encourage.

**Review by doomspork — APPROVED** [review]

**Pull request merged**